File: MasterErp.Angular/src/app/components/Purchases/models/PurchaseInvoiceModel.ts
import { OrderDetailModel } from "src/app/components/Shared/models/ItemModel";
import { GeneralOrderDetailsModel, GeneralOrderModel } from "../../Inventory/models/GeneralOrderModel ";



export interface PurchaseInvoiceModel extends GeneralOrderModel {
    purchaseInvoiceId: number | null;
    materialReceiptId: number;
    supplierId: number;
    supplierNameAR: string;
    supplierNameEN: string;
    orderDetails: GeneralOrderDetailsModel[];
    totalCount?: number;
    isChecked: boolean;

}

export interface PurchaseInvoiceDetails {
 
}

File: RetailStation.Entities/DTOs/Purchases/PurchaseInvoiceModel.cs
using RetailStation.Entities.DTOs.Inventory;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.DTOs.Purchases
{

    public class PurchaseInvoiceModel : GeneralOrderModel
    {
        public int? PurchaseInvoiceId { get; set; }
        public List<int> MaterialReceiptIds { get; set; }
        public int? MaterialReceiptId { get; set; }
        public int SupplierId { get; set; }
        public string SupplierNameAR { get; set; }
        public string SupplierNameEN { get; set; }
        public double? Discount { get; set; }
        public double? DiscountPercent { get; set; }
        public double? Tax { get; set; }
        public double? TaxPercent { get; set; }
        public double? NetValue { get; set; }
        public List<GeneralOrderDetailsModel> OrderDetails { get; set; }

        public int? TotalCount { get; set; }

    }
}
